### ISSUE TITLE
Encoder[BigDecimal] is not serializable

### ISSUE THREAD

**Issue by p4085f9** [sksamuel/avro4s#432]
We are using Avro4s to build Avro schemas.

After migrating the Avro version from 1.8.2 to 1.9.1 and avro4s 2.12:3.0.4 we faced an NotSerializableException: org.apache.avro.Conversions$DecimalConversion

The following code reproduces the issue:

```
import java.io.{ByteArrayOutputStream, ObjectOutputStream}
import com.sksamuel.avro4s.Encoder

val oos = new ObjectOutputStream(new ByteArrayOutputStream())
oos.writeObject(Encoder.bigDecimalEncoder)
oos.close
```

We assume Encoders should be serializable. The problem is that avro:DecimalConverion is not serializable which is out of scope of avro4s.

The following is a potential Bug-Fix for avro4s in line https://github.com/sksamuel/avro4s/blob/03556542ed8454b68cf64a9fd673d1338edb1425/avro4s-core/src/main/scala/com/sksamuel/avro4s/Encoder.scala#L245:
@transient private lazy val converter = new Conversions.DecimalConversion

Is that bugfix in line with the rest of the project? Should we open a PR or will you take it from the issue?

**Comment by sksamuel** [comment]
A PR would be good.

**Comment by p4085f9** [comment]
Thank you for your feedback. 
OK, I am going to prepare it.

**Comment by andreas-schroeder** [comment]
Fixed via #459
